pkg/query-service/app/clickhouseReader/reader.go

```diff
@@ -810,6 +824,22 @@ func (r *ClickHouseReader) GetServices(ctx context.Context, queryParams *model.G
 			defer func() { <-sem }()
 			var serviceItem model.ServiceItem
 			var numErrors uint64
+
+			// Even if the total number of operations within the time range is less and the all
+			// the top level operations are high, we want to warn to let user know the issue
+			// with the instrumentation
+			serviceItem.DataWarning = model.DataWarning{
+				TopLevelOps: (*allTopLevelOps)[svc],
```

Comment:
sending all the values of `allTopLevelOps` in field `DataWarning`? This is a lot of unwanted data?

Reply:
It's not unwanted. It is possible that In the last 30 minutes, the unique top-level ops may be well within the limit, however, overall customers sending operations that are cardinal enough to cause issues. We will show warning message even if the API works without an issue for the selected time but has underlying data problems.

Reply:
>but has underlying data problems
where is this logic?